libs/utils/android/workloads/uibench.py

```diff
@@ -0,0 +1,116 @@
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+
+from subprocess import Popen, PIPE
+from android import Screen, System, Workload
+from time import sleep
+
+import logging
+
+class UiBench(Workload):
+    """
+    Android UiBench workload
+    """
+
+    # Package required by this workload
+    package = 'com.android.test.uibench'
+
+    # Setup logger
+    logger = logging.getLogger('UiBench')
+    logger.setLevel(logging.INFO)
+
+
+    def __init__(self, test_env):
+        super(UiBench, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'UiBench')
+
+    def run(self, exp_dir, test_name, duration_s, collect=''):
+        activity = '.' + test_name + 'Activity'
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Press Back button to be sure we run the video from the start
+        System.menu(self.target)
+        System.back(self.target)
+
+        # Close and clear application
+        System.force_stop(self.target, self.package, clear=True)
+
+        # Set airplane mode
+        System.set_airplane_mode(self.target, on=True)
+
+        # Force screen in PORTRAIT mode
+        Screen.set_orientation(self.target, portrait=True)
+
+        # Clear logcat
+        os.system(self._adb('logcat -c'));
+
+        # Regexps for benchmark synchronization
+        start_logline = r'ActivityManager: START.*'\
+                         'cmp=com.android.test.uibench/{}'.format(activity)
+        UIBENCH_BENCHMARK_START_RE = re.compile(start_logline)
+        logging.debug("START string [%s]", start_logline)
+
+        # Parse logcat output lines
+        logcat_cmd = self._adb(
+                'logcat ActivityManager:* System.out:I *:S BENCH:*'\
+                .format(self.target.adb_name))
+        self.logger.info("%s", logcat_cmd)
+
+        # Start the activity
+        System.start_activity(self.target, self.package, activity)
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+        while True:
+
+            # read next logcat line (up to max 1024 chars)
+            message = logcat.stdout.readline(1024)
+
+            # Benchmark start trigger
+            match = UIBENCH_BENCHMARK_START_RE.search(message)
+            if match:
+                if 'energy' in collect and self.te.emeter:
+                    self.te.emeter.reset()
+                self.logger.debug("Benchmark started!")
+                break
+
+        # Run the workload for the required time
+        logging.info('Benchmark [%s] started, waiting %d [s]',
+                     activity, duration_s)
+        sleep(duration_s)
+
+        if 'energy' in collect and self.te.emeter:
+            nrg_data, nrg_file = self.te.emeter.report(exp_dir)
```

Comment:
Since https://github.com/ARM-software/lisa/commit/e21af00b704f15d26fc882555e21b7e7dec0ef0a, energy meters return a `namedtuple` that contains energy data and the path to the energy file. Hence, this should be changed to:

```python
nrg_report = self.te.emeter.report(exp_dir)
```